### PR TITLE
Allow to select the edit layout for subforms in custom fields

### PR DESCRIPTION
**Pull request by HLeithner**
### Summary of Changes
Add a column to the subform custom field which allows to select they layout for editing

### Testing Instructions
* create a subform
* select the type of the edit layout
* layout is used on editing

### Actual result BEFORE applying this Pull Request
Not possible

### Expected result AFTER applying this Pull Request
Works

### Documentation Changes Required
[possible
](https://docs.joomla.org/Help4.x:Fields:_Edit)

### CODE REVIEW

**Comment by brianteeman** [comment]
This is a good example of issues with the editorconfig that we now have.

The file is using tabs but your new code is using spaces :(

**Comment by HLeithner** [comment]
> This is a good example of issues with the editorconfig that we now have.
> 
> The file is using tabs but your new code is using spaces :(

hmm you are right, seems I was to optimistic for the editorconfig.

I will write a PR that only php files uses spaces.

**Comment by HLeithner** [comment]
Now I used tabs in the xml

File: administrator/language/en-GB/plg_fields_subform.ini [hunk 1]
```diff
@@ -10,4 +10,7 @@ PLG_FIELDS_SUBFORM_PARAMS_MAX_ROWS_LABEL="Maximum Rows"
 PLG_FIELDS_SUBFORM_PARAMS_OPTIONS_LABEL="Fields"
 PLG_FIELDS_SUBFORM_PARAMS_RENDER_VALUES_LABEL="Render Values"
 PLG_FIELDS_SUBFORM_PARAMS_REPEAT_LABEL="Repeatable"
+PLG_FIELDS_SUBFORM_PARAMS_EDIT_LAYOUT_LABEL="Edit Layout"
+PLG_FIELDS_SUBFORM_PARAMS_EDIT_LAYOUT_OPTION_REPEATABLE_TABLE_LABEL="Repeatable Table"
+PLG_FIELDS_SUBFORM_PARAMS_EDIT_LAYOUT_OPTION_REPEATABLE_FORM_LABEL="Repeatable Form"
```
Comment (Quy) [inline administrator/language/en-GB/plg_fields_subform.ini]: Please alpha sort keys.

**Comment by crystalenka** [comment]
Works well, and is a need! Just would recommend moving the field to Options tab in the Form Options field group, as that pattern is already established in other field types like radio:

**Comment by crystalenka** [comment]
Hahaha my change made something not work. Sorry. 🙈 I'll open another PR against your branch after Amelia is in bed and I can fix it.

**Comment by Hackwar** [comment]
Since 4.2 only accepts Bugfixes and this is a new feature (or rather an improvement), I've rebased the PR to 4.3-dev and updated it.

File: administrator/language/en-GB/plg_fields_subform.ini (hunk 1 above)
Comment (Hackwar) [reply administrator/language/en-GB/plg_fields_subform.ini]: Done

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on f25d7594556e945d5b3e553401fbd4b35e777e8a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38180">issues.joomla.org/tracker/joomla-cms/38180</a>.</sub>

**Comment by carlitorweb** [comment]
I have tested this item :white_check_mark: successfully on f25d7594556e945d5b3e553401fbd4b35e777e8a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38180">issues.joomla.org/tracker/joomla-cms/38180</a>.</sub>

**Comment by carlitorweb** [comment]
Documentation added :heavy_check_mark: 
https://docs.joomla.org/Help4.x:Fields:_Edit

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38180">issues.joomla.org/tracker/joomla-cms/38180</a>.</sub>

**Comment by HLeithner** [comment]
Thanks @carlitorweb for the documentation update

**Pull request merged**